tfe/resource_tfe_policy.go

```diff
@@ -0,0 +1,308 @@
+package tfe
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceTFEPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEPolicyCreate,
+		Read:   resourceTFEPolicyRead,
+		Update: resourceTFEPolicyUpdate,
+		Delete: resourceTFEPolicyDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceTFEPolicyImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the policy",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+
+			"description": {
+				Description: "Text describing the policy's purpose",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+
+			"organization": {
+				Description: "Name of the organization that this policy belongs to",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+
+			"kind": {
+				Description: "The policy-as-code framework for the policy. Valid values are sentinel and opa",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     string(tfe.Sentinel),
+				ValidateFunc: validation.StringInSlice(
+					[]string{
+						string(tfe.OPA),
+						string(tfe.Sentinel),
+					}, false),
+			},
+
+			"query": {
+				Description: "The OPA query to run. Required for OPA policies",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+
+			"policy": {
+				Description: "Text of a valid Sentinel or OPA policy",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+
+			"enforce_mode": {
+				Type: schema.TypeString,
+				Description: fmt.Sprintf(
+					"The enforcement configuration of the policy. For Sentinel, valid values are %s. For OPA, Valid values are `%s`", sentenceList(
+						sentinelPolicyEnforcementLevels(),
+						"`",
+						"`",
+						"and"),
+					sentenceList(
+						opaPolicyEnforcementLevels(),
+						"`",
+						"`",
+						"and")),
+				Optional: true,
```

Comment:
I don't think this field is truly optional. I got this error when omitting it:

```
╷
│ Error: Error creating sentinel policy my-sentinel-policy for organization hashicorp: invalid attribute
│
│ invalid format for enforce
│
│   with tfe_policy.foo-sentinel,
│   on main.tf line 6, in resource "tfe_policy" "foo-sentinel":
│    6: resource "tfe_policy" "foo-sentinel" {
│
```

Reply:
Good catch! 👍  Turns out the [api docs](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/policies) around `data.attributes.enforce[].mode`  are wrong. The enforce_mode is `required` in Atlas.

I have made changes to add defaults if the enforce mode is not provided. 
Since the `tfe_sentinel_policy` provider sets the enforce_mode as [optional](https://github.com/hashicorp/terraform-provider-tfe/blob/main/tfe/resource_tfe_sentinel_policy.go#L50), I have done the same for `tfe_policy`, so our customers dont have to care about the enforce_mode when they switch from `tfe_sentinel_policy` to `tfe_policy` and the experience is the same.

Meanwhile I am going to go fix the API docs :)